com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs

```diff
@@ -478,7 +510,56 @@ string RendererDataDisplayName(ScriptableRendererData data)
             return "NULL (Missing RendererData)";
         }
 
+        internal bool CanRemoveFromRendererDataList(ScriptableRendererData rendererData)
+        {
+            // Check if it exists and it is not the default renderer data
+            return m_RendererDataList.Contains(rendererData) && rendererData != scriptableRendererData;
+        }
+
+        internal bool RemoveRendererFromRendererDataList(ScriptableRendererData rendererData)
+        {
+            // Check if it exists
+            if (CanRemoveFromRendererDataList(rendererData))
+            {
+                int index = RendererDataIndex(rendererData);
+                if (index != -1)
+                {
+                    UpdateDefaultRendererValue(index);
+                }
+
+                // Lets remove the renderer data
+                var rendererDataList = new List<ScriptableRendererData>(m_RendererDataList);
```

Comment:
If the index is -1 you do not need to copy and delete, besides that you can simply call:

`rendererDataList.RemoveAt(index)`, and avoid another lookup for the renderer.